.ci/run.py

```diff
@@ -96,6 +99,48 @@ def print_build(script):
     writeln_console("================================================================")
 
 
+@contextmanager
+def ensure_cache_preserved():
```

Comment:
clever solution! I would use git init + git add + git diff to check any different, but your solution works pretty well :)

Reply:
I think I like yours more. Easier to implement, cleaner... it is just much better. Thanks!

Reply:
oh, much simpler now!